File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSOperationalPoint.java
package fr.sncf.osrd.railjson.schema.infra;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import fr.sncf.osrd.railjson.schema.common.Identified;
import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSOperationalPointPart;
import java.util.List;
import org.jetbrains.annotations.Nullable;

@SuppressFBWarnings({"UWF_UNWRITTEN_PUBLIC_OR_PROTECTED_FIELD"})
public class RJSOperationalPoint implements Identified {
    public String id;
    public List<RJSOperationalPointPart> parts;

    @Nullable
    public RJSOperationalPointExtensions extensions;

    @Nullable
    public String weight;

    @Nullable
    public String plc;

    public RJSOperationalPoint(
            String id,
            List<RJSOperationalPointPart> parts,
            @Nullable RJSOperationalPointExtensions extensions,
            @Nullable String weight,
            @Nullable String plc) {
        this.id = id;
        this.parts = parts;
        this.extensions = extensions;
        this.weight = weight;
        this.plc = plc;
    }

    @Override
    public String getID() {
        return id;
    }
}

File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/infra/RJSOperationalPoint.kt
package fr.sncf.osrd.railjson.schema.infra

import com.squareup.moshi.Json
import fr.sncf.osrd.railjson.schema.common.Identified
import fr.sncf.osrd.railjson.schema.infra.trackobjects.RJSOperationalPointPart

class RJSOperationalPoint(
    override val id: String,
    val parts: List<RJSOperationalPointPart>,
    val weight: Long?,
    val name: String,
    val uic: Long?,
    val plc: String?,
    @Json(name = "country_code") val countryCode: String,
    @Json(name = "main_code") val mainCode: String,
    @Json(name = "secondary_code") val secondaryCode: String?,
    @Json(name = "is_passenger_station") val isPassengerStation: Boolean,
    @Json(name = "secondary_name") val secondaryName: String?,
) : Identified
